### PR TITLE
Substitute stations

### PR DESCRIPTION
**Pull request by juliayyan**
- The method `substitutestations()` to compute a station's substitutes.  A station `u1` is substitutable for another station `u` if (a) `u` and `u1` are not on the same line (to prevent uninteresting substitutions), and (b) `u` and `u1` are at most `maxdist` away from each other.
- The method `expandedstages()` to compute a variety of different ways to get from `u` to `v`.  A wider variety of route choices can be generated by looking not just at `(u,v)`, but also stations that are substitutable for `u` and `v`.
- The associated tests are in `test/multiroutes-construction.jl`

### CODE REVIEW

File: src/network/networkmethods.jl [hunk 1]
```diff
@@ -199,6 +199,23 @@ publicchoices(np::TransitNetworkProblem, u::Int, v::Int) = choices(np,u,v,false)
...
+            isassigned(np.stage, usub, vsub) || continue 
+            (usub == u) && (vsub == v) && continue
+            for r in 1:length(np.stage[usub, vsub])
+                push!(od, (usub, vsub))
+                push!(routes, np.stage[usub, vsub][r])
+            end 
+        end 
+    end 
+    return od, routes
+end
```
Comment (yeesian) [inline src/network/networkmethods.jl]: """
Returns the tuple `(od,routes)` where
* `od` is a vector of (possibly duplicate) origin-dest pairs
* `routes` is a vector of stage corresponding to a unique route for the corresponding `od` pair
```
"""

File: src/network/networkmethods.jl [hunk 2]
```diff
@@ -227,3 +244,15 @@ function identifylines(np::TransitNetworkProblem, u::Int, v::Int, r::Int)
...
+    stns = Int[]
+    dists = [haversinedistance(np,u,v) for v in 1:np.nstations]
+    separated = zeros(Bool, np.nstations)
+    for v in 1:np.nstations 
+        u == v && continue
+        length(intersect(np.platforms[u], np.platforms[v])) > 0 && continue 
+        separated[v] = true
+    end 
+    return (1:np.nstations)[separated .& (dists .<= maxdist)]
+end
```
Comment (yeesian) [inline src/network/networkmethods.jl]: "Returns stops within `maxdist` of `u` that do not correspond to the same parent station."

**Comment by yeesian** [comment]
The codecov and coveralls reports seem to be reporting the same thing, and is kind of noisy.. Might be good to silence one of them, haha.

**Comment by juliayyan** [comment]
Okay, I added the docstrings.

Do you have a preference for which of codecov / coveralls you'd rather keep?

**Comment by yeesian** [comment]
> Do you have a preference for which of codecov / coveralls you'd rather keep?

I don't have a preference; let's go with codecov for now?

**Comment by yeesian** [comment]
Looks good to me now, so I'll just merge this. Thanks!

**Pull request merged**